scripts/cleanup.js

```diff
@@ -53,124 +67,116 @@ function getImages(region, repo, callback) {
     ecr.describeImages(params, (err, data) => {
       if (err) return callback(err);
       details = details.concat(data.imageDetails);
-      data.nextToken ? describeImages(repo, data.nextToken, callback) : callback(null, details);
+
+      if (!data.nextToken) return callback(null, details);
+
+      describeImages(repo, data.nextToken, callback);
     });
   }
 
 }
 
 module.exports.imagesToDelete = imagesToDelete;
-function imagesToDelete(images, callback) {
+function imagesToDelete(images, tmpdir, callback) {
   const q = queue(1);
-  images = images.sort((a, b) => { return (new Date(a.imagePushedAt) - new Date(b.imagePushedAt));
-  });
+  const generic = [];
+  const priority = [];
 
-  let cruftDigests = [];
-  let deployDigests = [];
-
-  for(let img of images) {
-    q.defer(commitType, img.imageTags[0], img.imageDigest);
-  }
+  // Sort oldest to newest, categorize each image
+  images
+    .sort((a, b) => new Date(a.imagePushedAt) - new Date(b.imagePushedAt))
+    .forEach((img) => q.defer(commitType, img.imageTags[0], img.imageDigest, tmpdir));
 
   q.awaitAll((err, data) => {
     if (err) return callback(err);
 
-    for (let d of data) {
-      for (let digest in d) {
-        if (d[digest] === 'commit') cruftDigests.push({ imageDigest: digest });
-        else if (d[digest] !== 'custom') deployDigests.push({ imageDigest: digest });
-      }
-    }
+    // Select generic images and priority images
+    data.forEach((result) => {
+      if (result.type === 'generic')
+        generic.push({ imageDigest: result.digest });
+      if (result.type === 'priority')
+        priority.push({ imageDigest: result.digest });
+    });
 
-  });
+    // We want to leave the repository with MAX_IMAGES number of images in it.
+    let imagesToDelete;
+    let excessImages = images.length - MAX_IMAGES;
```

Comment:
These two variables are god examples of where you should use `let`. You should prefer `const` over let if you are not going to actually change the reference.

Further along here, you see that there are several statements like

```
imagesToDelete = some new object for this variable to reference
```

```
excessImages = excessImages - some value
```

Only when you're changing what the variable will reference by setting it `=` a new thing should you use `let`.